python/task_manager/background_task_manager.py

```diff
@@ -1,47 +1,142 @@
 from sgtk import TankError
 
 from .background_task import BackgroundTask
 from .worker_thread import WorkerThread
 
+
 class BackgroundTaskManager(QtCore.QObject):
     """
     Main task manager class. Manages a queue of tasks running them asynchronously through
     a pool of worker threads.
 
-    Note that the BackgroundTaskManager class itself is reentrant but not thread-safe so its methods should only 
+    Note that the BackgroundTaskManager class itself is reentrant but not thread-safe so its methods should only
     be called from the thread it is created in. Typically this would be the main thread of the application.
-    
-    
+
+
     :signal task_completed(uid, group, result): Emitted when a task has been completed.
-        The ``uid`` parameter holds the unique id associated with the task, 
-        the ``group`` is the group that the task is associated with and 
-        the ``result`` is the data returned by the task. 
-    
+        The ``uid`` parameter holds the unique id associated with the task,
+        the ``group`` is the group that the task is associated with and
+        the ``result`` is the data returned by the task.
+
     :signal task_failed(uid, group, message, traceback_str): Emitted when a task fails for some reason.
-        The ``uid`` parameter holds the unique id associated with the task, 
-        the ``group`` is the group that the task is associated with, 
+        The ``uid`` parameter holds the unique id associated with the task,
+        the ``group`` is the group that the task is associated with,
         the ``message`` is a short error message and the ``traceback_str``
         holds a full traceback.
-        
+
     :signal task_group_finished(group): Emitted when all tasks in a group have finished.
         The ``group`` is the group that has completed.
 
     """
 
+    class _ResultsPoller(QtCore.QObject):
+        """
+        Polls a queue which holds the results posted from the worker threads.
+        """
+
+        # Emitted when a task is completed.
+        task_completed = QtCore.Signal(object, object, object)
+        # Emitted when a task has failed.
+        task_failed = QtCore.Signal(object, object, object, object)
+
+        def __init__(self, parent=None):
+            """
+            Constructor.
+
+            :param parent:  The parent QObject for this thread
+            """
+            QtCore.QObject.__init__(self, parent)
+            # Results queue that will be polled
+            self._results = Queue.Queue()
+            # Create a timer with no interval. This means the timer will be invoked
+            # only when the event queue is empty.
+            self._timer = QtCore.QTimer()
+            self._timer.setInterval(100)
+            # This should be instantiated from the application's thread since we only support
+            # one instance for now.
+            assert(QtCore.QCoreApplication.instance().thread() == self._timer.thread())
```

Comment:
This feels very C++! This would prevent anything from using this outside the main thread right? I don't think that's what we want. The way I see this is that we have a bug fix that helps with stabilty *if* the task mgr is executed from the main thread. Our fix potentially doesn't fix issues if the task manager is run from a worker thread (but may still reduce the frequency of them) but I don't think that means that we should prevent people from using it. Instead, I think we should update the docs with an info/warning box/blurb to highlight the fact that there *may* be issues on some combinations on pyside/python if you run it from a thread.